### ISSUE TITLE
Space after italics gets deleted when using keyboard shortcut

### ISSUE THREAD

**Issue by killermann** [WordPress/gutenberg#12529]
**Describe the Bug**
When writing, if you use keyboard shortcut for italics (or bold), then hit space before un-italicizing, the space gets deleted when you un-italicize, and any further typing gets italicized and combined to the previous word.

**Steps to reproduce the behavior:**
1. Start typing in a text block
2. Use keyboard shortcut to switch to italics (cmd+i)
3. Type italicized word, then a space
4. Use keyboard shortcut to cancel italics (cmd+i)
5. Type the next word in sentence.

**Expected behavior**
Ideally, the editor would put a closing </em> tag on the last character before the space, and allow you to keep typing un-italicized (or un-bolded, etc.). 

At the very least, there should be a non-breaking space after the character, then the closing </em> tag.

In either case, you would be able to continue writing un-interrupted. Right now, it requires you to use the mouse to fix, or the right arrow key, which is disruptive to the user's writing flow.

**Screenshots**
Here's a gif of a short video of the behavior: https://imgur.com/iYvzfBG

I hope it helps, because this was a weird one to describe.

**Desktop (please complete the following information):**
 - OS: MacOS Mojave 10.14.1
 - Browser: Chrome
 - Version 70.0.3538.110

**Additional context**
- Gutenberg 4.6.1
- Replicated on latest WP, as well as latest 5.0RC

**Comment by torres126** [comment]
Automattic/wp-calypso/issues/#28834

**Comment by torres126** [comment]
Automattic/wp-calypso#28834

**Comment by sirjonathan** [comment]
I was having the same issue _yesterday_ and came here today to report it. On testing again today, though, I was not able to reproduce the issue in either the latest versions of Firefox or Chrome. I'm running WordPress 5.0.2 and the Gutenberg plugin is deactivated.

@killermann - I'm curious if you can reproduce the issue again?

**Comment by torres126** [comment]
I can confirm that this can still be reproduced with Gutenberg on WordPress.com.

**Comment by killermann** [comment]
@sirjonathan just give it a go in Chrome with WP 5.0.2 (not Gutenberg plugin). Identical behavior to my screen recordings above. 

So, on my end, still a bug.

One change: I'm not seeing all the &nbsp; spam in the output, so that's progress.

(sorry for the delay -- holidays and whatnot)

**Comment by AtrumGeost** [comment]
Hi folks!

I was able to reproduce this using a Jurassic Ninja site and it also happens when you use the "Bold" button. Here's what I got:

![screen capture on 2019-01-28 at 12-56-07](https://user-images.githubusercontent.com/3696121/51855813-26d35d80-22fc-11e9-8a40-63a89951d455.gif)

Since the space is deleted, the text will go back to bold. I wasn't able to reproduce this in the Classic Editor.

_WordPress 5.0.3 running Twenty Nineteen theme. Chrome Version 71.0.3578.98 (Official Build) (64-bit). macOS Mojave 10.14.2_
